features/admin.impersonation.v1/pages/impersonation-configuration.tsx

```diff
@@ -0,0 +1,277 @@
+import { addAlert } from "@wso2is/core/store";
+import { Field, Form, FormPropsInterface } from "@wso2is/form";
+import { EmphasizedSegment, PageLayout } from "@wso2is/react-components";
+import React, { FunctionComponent, MutableRefObject, ReactElement, useEffect, useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider, Grid, Placeholder, Ref } from "semantic-ui-react";
+import {
+    updateImpersonationConfigurations,
+    useImpersonationConfig
+} from "../api/impersonation-configuration";
+import {
+    ImpersonationConfigFormValuesInterface,
+    ImpersonationConfigPatchInterface
+} from "../models/impersonation-configuration";
+
+/**
+ * Props for impersonation Configuration settings page.
+ */
+type ImpersonationConfigurationPageInterface = IdentifiableComponentInterface;
+
+const FORM_ID: string = "impersonation-configuration-form";
+
+/**
+ * Impersonation Configuration page.
+ */
+export const ImpersonationConfigurationPage: FunctionComponent<ImpersonationConfigurationPageInterface> = (
+    props: ImpersonationConfigurationPageInterface
+): ReactElement => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const pageContextRef: MutableRefObject<any> = useRef(null);
+    const formRef: MutableRefObject<FormPropsInterface> = useRef<FormPropsInterface>(null);
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+
+    const isReadOnly: boolean = useMemo(() => !hasRequiredScopes(
+        featureConfig?.server,
+        featureConfig?.server?.scopes?.update,
+        allowedScopes
+    ), [ featureConfig, allowedScopes ]);
+
+    const dispatch: Dispatch<any> = useDispatch();
+
+    const { t } = useTranslation();
+
+    const [ impersonationConfig, setImpersonationConfig ] =
+        useState<ImpersonationConfigFormValuesInterface>(undefined);
+
+    const {
+        data: originalImpersonationConfig,
+        isLoading: isImpersonationFetchRequestLoading,
+        mutate: mutateImpersonationConfig,
+        error: impersonationConfigFetchRequestError
+    } = useImpersonationConfig();
+
+    useEffect(() => {
+        if (originalImpersonationConfig instanceof IdentityAppsApiException
+            || impersonationConfigFetchRequestError) {
+            handleRetrieveError();
+
+            return;
+        }
+
+        if (!originalImpersonationConfig) {
+            return;
+        }
+
+        setImpersonationConfig({
+            enableEmailNotification: originalImpersonationConfig.enableEmailNotification
+        });
+    }, [ originalImpersonationConfig ]);
+
+    /**
+     * Displays the error banner when unable to fetch Impersonation configuration.
+     */
+    const handleRetrieveError = (): void => {
```

Comment:
Since this is not reused anywhere, let's directly use addAlert function in respective places instead of defining a new function